### PR TITLE
!env support in config

### PR DESCRIPTION
**Pull request by yannick**
implements https://github.com/h2o/h2o/issues/1520 , replaces #1519 

- [x] documentation

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1393,6 +1393,27 @@ static yoml_t *resolve_file_tag(yoml_t *node, resolve_tag_arg_t *arg)
...
+        fprintf(stderr, "value of !env must be a scalar");
+        return NULL;
+    }
+
+    const char *env_var_content;
+    if ((env_var_content = getenv(env_var)) == NULL)
+        env_var_content = "";
+
+    free(node->data.scalar);
+    node->data.scalar = h2o_strdup(NULL, env_var_content, SIZE_MAX).base;
```
Comment (yannick) [inline src/main.c]: what is the reason the pointer from `getenv` cant be assigned directly to `node->data.scalar` ?

Comment (kazuho) [reply src/main.c]: Because `node->data.scalar` eventually gets `free`d in `yoml_free`.

**Comment by yannick** [comment]
Q: should we allow a default value? if so how, just `strstr` for the first space?

**Comment by kazuho** [comment]
@yannick 
> Q: should we allow a default value? if so how, just `strstr` for the first space?

Could you please elaborate what you mean by "default"? To me the current code that uses an empty string upon the absence the environment variable seems sufficient to me.

**Comment by yannick** [comment]
@kazuho one could catch the null and then set the variable to a default that was supplied. Like `!env H2O_PORT 8080`

If you think thats useless im fine as is.

**Comment by yannick** [comment]
Use case for me would be to simplify deployment of mini services in containers

**Comment by kazuho** [comment]
@yannick Sorry for the belated response.

> Like `!env H2O_PORT 8080`.

We could do that, but the syntax could be an issue. Personally I'd prefer using a syntax similar to bash (e.g., `!env "H2O_ROOT:-8080"`).

What do you think? Alternatively, we could leave default value support to another PR.

**Comment by yannick** [comment]
@kazuho i am fine with this idea. i'd say we leave it as is and if i manage i'll do a 2nd pr to add defaults.

**Comment by kazuho** [comment]
Thank you for working on the PR. The changes have been merged to master.

**Pull request merged**